### PR TITLE
Quick and dirty implementation of container logs

### PR DESCRIPTION
**Pull request by robertsheehy-wf**
screen shot 2015-04-30 at 12 48 57 am

### CODE REVIEW

**Comment by robertsheehy-wf** [comment]
@ngmiller meh, probably not the best solution, but it's quick and dirty

File: app/js/instances/containerController.js [hunk 1]
```diff
@@ -19,20 +19,47 @@
...
+    var loadLogs = function() {
+        dockerService.d('containers.logs', {
+            host: $scope.host,
+            id: $scope.id,
+            responseType: 'text',
+            query: {
+                stderr: '1',
+                stdout: '1',
+                timestamps: '0',
+                follow: '0'
```
Comment (ngmiller) [inline app/js/instances/containerController.js]: #nitpick These values can be booleans. $http will serialize them properly

**Comment by ngmiller** [comment]
Looks good. The only issue seems to be redundant logs in the responses. I set up a container to ping google and get the entire log back on every subsequent request when "follow" is checked, meaning the response time increases with every request :fearful: 

Unfortunately, Docker doesn't seem to provide any help. `/containers/(id)/attach` seems to be overkill for just output and also encodes the response in a custom format so it can multiplex stdout and stderr.

Were you able to get streaming working? If so, I think we could just automatically dispatch each log response to a model and follow the stream by default. But then there's the issue of a user wanting to stop the stream. If they stop it for a while then start it again, the simplest thing to do is reload the entire log stream again, but that could be huge. (At least for real applications serving lots of requests)

Have you come across any information with regards to log "pagination"?

**Comment by zach-taylor** [comment]
Works for me. +1

**Comment by robertsheehy-wf** [comment]
@ngmiller 
> Were you able to get streaming working?

No

> Have you come across any information with regards to log "pagination"?

Right now I don't think pagination is possible. The only way to truncate from docker is to use the tail argument which only outputs the n latest logs.

<br>
I agree this isn't a long term solution :disappointed:, especially since logs can be humungous.  For the time being we can probably look at putting a hard limit on how many logs we pull by setting tail to ~10000.

**Comment by ngmiller** [comment]
@robertsheehy-wf Oh man I didn't even see the tail parameter. Would `?tail=1` on the "follow" option work?

**Comment by ngmiller** [comment]
Yeah, I think getting a long-term solution to this is outside of the scope of work in terms of the time we have left. Works for me! +1

**Pull request merged**